language-design.md

```diff
@@ -17,6 +17,97 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+*“According to findings by researchers from Southern Illinois University, this 
+reaction isn’t just because you’re a n00b: they found that Perl, a major programming 
+language used by untold zillions of developers, is no more intuitive to novices than a 
+language with a randomly generated syntax. Why shouldn’t those "interfaces" be humanely 
+designed? … They created a "placebo language" called Randomo, whose syntax was randomly 
+generated, to use in trials alongside Quorum and Perl. Novice programmers were able to write 
+sample programs more accurately in Quorum versus Perl — an interesting, but not terribly 
+surprising, result. More surprising was how Perl compared to Randomo. To quote the paper: 
+'Perl users were unable to write programs more accurately than those using a language designed 
+by chance.' … Every piece of software we use was written by other people, slaving away over 
+thousands of lines of code. Why shouldn’t those "interfaces" be as humanely designed as 
+the ones we tap and swipe?”* 
+
+This quote really resonated with me because my first experience with “programming” was with 
+MIT app inventor. Even as a middle schooler, I never had to look at documentation to 
+understand how to “write code” in app inventor. Each piece of code was represented in the 
+form of a “puzzle” piece:
```

Comment:
One of the things I found quite interesting about these languages was the impossibility of syntax errors, blocks simply could not fit together if they shouldn't.  The really interesting thing for me is how the design of languages leads to the treatment of syntax vs semantic errors.  In languages typically used by newer programmers, it seems the programming language might be more lenient (preventing too many syntax errors from overwhelming a newcomer), but at the cost of turning some mistakes into semantic errors (which will be caught at runtime, or not at all).

This seems like a good tradeoff, since (in my experience grutoring at least) it is easier to convince new programmers to keep a positive outlook when their code _runs_ but does the wrong thing, while it is very often discouraging to be unable to even run one's code (for example when trying to help a student with a dozen compiler *and* linker errors).

However, when one gains more familiarity and fluency in the culture of programming, it is often desired (as we can see from the value judgements of our DSL class of experienced coders) for a language to fail early and loudly, just as languages like C++ and Haskell do, enforcing strict rules and type hierarchies to avoid running one thing when we meant another.

I'm curious if others agree or disagree with me that the fail early + fail loudly quality of languages is not actually an inherent strength of a language, but rather a trade-off like many others which makes the language more useful to experienced coders, but less friendly to newer members of the community.  Thoughts?  :)

Also, I googled "Syntax Error Scratch", trying to find a reference or picture of blocks not fitting together, but instead I found [this](https://scratch.mit.edu/projects/76344642/).